src/BlockBuilderPolicy.sol

```diff
@@ -174,35 +190,52 @@ contract BlockBuilderPolicy is Initializable, UUPSUpgradeable, OwnableUpgradeabl
     function isAllowedPolicy(address teeAddress) public view returns (bool allowed, WorkloadId) {
+        (bool isValid, FlashtestationRegistry.RegisteredTEE memory registration) =
+            FlashtestationRegistry(registry).getRegistration(teeAddress);
+        if (!isValid) {
+            return (false, WorkloadId.wrap(0));
+        }
+
+        WorkloadId workloadId = workloadIdForTDRegistration(registration);
         for (uint256 i = 0; i < workloadIds.length(); ++i) {
-            WorkloadId workloadId = WorkloadId.wrap(workloadIds.at(i));
-            if (FlashtestationRegistry(registry).isValidWorkload(workloadId, teeAddress)) {
+            if (WorkloadId.unwrap(workloadId) == workloadIds.at(i)) {
                 return (true, workloadId);
             }
         }
+
         return (false, WorkloadId.wrap(0));
     }
 
-    /// @notice An alternative implementation of isAllowedPolicy that verifies more than just
-    /// the workloadId's matching and if the attestation is still valid
-    /// @param teeAddress The TEE-controlled address
-    /// @param expectedTeeTcbSvn The expected teeTcbSvn of the TEE's attestation
-    /// @return allowed True if the TEE's attestation is part of the policy, is still valid, and
-    /// the teeTcbSvn matches the expected value
-    /// @dev This exists to show how different Policies can be implemented, based on what
-    /// properties of the TEE's attestation are important to verify.
-    function isAllowedPolicy2(address teeAddress, bytes16 expectedTeeTcbSvn) external view returns (bool allowed) {
-        for (uint256 i = 0; i < workloadIds.length(); ++i) {
-            WorkloadId workloadId = WorkloadId.wrap(workloadIds.at(i));
-            TD10ReportBody memory reportBody = FlashtestationRegistry(registry).getReportBody(teeAddress);
-            if (
-                FlashtestationRegistry(registry).isValidWorkload(workloadId, teeAddress)
-                    && reportBody.teeTcbSvn == expectedTeeTcbSvn
-            ) {
-                return true;
-            }
-        }
-        return false;
+    // Application specific mapping of registration data to a workload identifier
+    // Think of the workload identifier as the version of the application for governance
+    // The workload id verifiably maps to a version of source code for the VM image
+    function workloadIdForTDRegistration(FlashtestationRegistry.RegisteredTEE memory registration)
+        public
+        pure
+        returns (WorkloadId)
+    {
+        bytes8 fullBitmask = 0xFFFFFFFFFFFFFFFF;
+        // We expect fpu and sse xfam bits to be set, and anything else should be handled by explicitly allowing the workloadid
+        bytes8 xfamExpectedBits = (fullBitmask ^ (TD_XFAM_FPU | TD_XFAM_SSE));
+
+        // We don't mind ve disabled and pks tdattributes bits being set either way, anything else requires explicitly allowing the workloadid
```

Comment:
your comment only includes 2 of the attributes, but the code line below includes 3 (VE_DISABLED, PKS, and KL). Should the comment be updated or the code? Either way, please make the code and comment align, and be correct.